### PR TITLE
Fixes #513: Renamed deleter methods to destroy

### LINKED ISSUE #513: NovelRT C API deleter methods are inconsistently named.
**_Note: for support questions, please use the #engine-user-help channel in our [Discord](https://discord.novelrt.dev/) or [create a discussion](https://github.com/novelrt/NovelRT/discussions/new). This repository's issues are reserved for feature requests and bug reports._**

**Describe the issue:**
There is a naming inconsistency in the C API where some of the deleter methods are tagged as `Delete` and some as `Destroy`.

**Please provide the steps to reproduce if possible:**
1. Clone the repo
2. Inspect the interop header files
3. Observe the issue

**Expected behaviour:**
 I believe the intended naming convention is `Destroy`.

**Please tell us about your environment:**  
N/A

**Additional context:**
This doesn't functionally break anything, its just gross to read.

### PR DESCRIPTION
**Pull request by Sakshi-75**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/novelrt/NovelRT/blob/misc/templates/Contributing.md#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR renames deleter methods in the interop by replacing delete word with destroy to maintain consistent method naming throughout.

**Is there an open issue that this resolves? If so, please [link it here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).**
Resolves #513 

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:

### CODE REVIEW

File: src/NovelRT.Interop/Maths/NrtQuadTreePoint.cpp
Comment (RubyNova) [inline src/NovelRT.Interop/Maths/NrtQuadTreePoint.cpp:39]:
```suggestion
    NrtResult Nrt_QuadTreePoint_Destroy(NrtQuadTreePointHandle point)
```

File: src/NovelRT.Interop/SceneGraph/NrtScene.cpp
Comment (RubyNova) [inline src/NovelRT.Interop/SceneGraph/NrtScene.cpp:57]:
```suggestion
    NrtResult Nrt_Scene_Destroy(NrtSceneHandle scene)
```

File: src/NovelRT.Interop/SceneGraph/NrtSceneNode.cpp
Comment (RubyNova) [inline src/NovelRT.Interop/SceneGraph/NrtSceneNode.cpp:203]:
```suggestion
    NrtResult Nrt_SceneNode_Destroy(NrtSceneNodeHandle node)
```

File: src/NovelRT.Interop/SceneGraph/NrtSceneNode.cpp
Comment (RubyNova) [inline src/NovelRT.Interop/SceneGraph/NrtSceneNode.cpp:223]:
```suggestion
    NrtResult Nrt_SceneNodeSet_Destroy(NrtSceneNodeSetHandle nodeSet)
```

**Review by RubyNova — CHANGES_REQUESTED** [review]
Mostly correct, just a few tweaks.

File: include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h
Comment (RubyNova) [inline include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h:22]: Not sure why GitHub deleted my comment on this one - this can be reverted since it reflects a member method in C++.

**Review by RubyNova — CHANGES_REQUESTED** [review]

File: include/NovelRT.Interop/Ecs/NrtCatalogue.h
Comment (Sakshi-75) [reply include/NovelRT.Interop/Ecs/NrtCatalogue.h]: Hi @RubyNova ,
I reverted this name. Can you review again please?

Comment (RubyNova) [reply include/NovelRT.Interop/Ecs/NrtCatalogue.h]: Can do. 2m please.

**Review by RubyNova — CHANGES_REQUESTED** [review]
Outstanding changes still exist from the first review. Once those are in I'll approve. :)

**Review by RubyNova — APPROVED** [review]
LGTM! Thank you for your contribution!

**Comment by Sakshi-75** [comment]
Hi @RubyNova ,
I addressed all the PR comments. Can you check if anything got missed?
Thanks!

**Comment by RubyNova** [comment]
All that's really missing is updating the branch with main since something else got merged since this PR was opened. I've enabled auto merge but it won't merge until:

- CI is fully passing
- this branch is up to date with main

The contents of the PR itself is fine. :)

**Pull request merged**
